Repair tests

StephanOpfer opened the pull request.

Commit:
added a functionality to acquire agentid and default arg to constructor declarations

Commit:
added changes after testing todo: code cleanup

Commit:
Removed ROS_INFO statements and ros.h

Commit:
- made the alica_tests compile under json format
- current issue: initialisation order of ALICA Engine -> things get messy, need a more proper solution here...

Commit:
- resolved circular header dependency between AlicaEngine.h and VariableSyncModule.h
- standardised include guard for classes under src/engine/expressionhandler/

Commit:
- fixed initialisation dependency
- renamed method in alica context

Commit:
- fix all but synchronisation test

Commit:
- renamed robotID to agentID
- improved debug output

Commit:
- tried to fix synchronisation module: Test works, but synch protocol/module seems to be completely broken

Commit:
- put debug output into canonical alica_debug_message() macros
- put a mutex in the destructor of SyncProcess

Commit:
- minor adaption of debug output

Commit:
- removed debug output from AlicaContext
- improved debug output in Synchronisation Process
- fixed multiple entries in SyncRow's ReceivedBy list
- fixed segfault/freez/undefined behaviour after a successfull synchronisation (deletion of SynchProcesses is now always done in SyncModule.tick())

Commit:
- removed unnecessary debug output

Commit:
- try to fix travis.yaml for pull requests that do not target rr-devel

Commit:
test again

Commit:
- travis.yaml: try to fix command to be compatible in one line

Commit:
- another try travis.yaml

Commit:
Revert "test again"

This reverts commit 7e1e4898463db2e083df51fe66f1785199ab60b8.

Commit:
- added some semicolons for single line command (wip)

Commit:
- trying to fix travis.yml with external shell script

Commit:
- forgot params

Commit:
- externalise reading lines

Commit:
- relative path

Commit:
- test for pwd

Commit:
- add missing $

Commit:
- fix for correct path of rosdep file

Commit:
- if string without quotes

Commit:
- add missing ;

Commit:
- add missing dependency libyaml-cpp-dev

Commit:
- adapted time for tests

Commit:
- added layout file for attack plan

Commit:
- improved script for not executing wstool up several times

Commit:
- increased available time for test execution from 60 to 300 sec

Commit:
changed constructor's agent parameter to AgentID* and copied the given AgentID* via the _agentIDManager

Commit:
- made agentID a const AgentID*
- fixed segfault in the list of initialisers of the AlicaEngine constructor (horrible line for TeamManager gets removed, when IDManager is owned by AlicaContext in future)

abhi1287 committed:
Merge pull request #84 from rapyuta-robotics/addAgentIdv0.9.0

Add agent idv0.9.0

veerajsk committed:
Take AgentID argument as an object instead of a pointer

veerajsk committed:
Take a const reference to agent id instead of the object itself

abhi1287 committed:
Merge pull request #104 from rapyuta-robotics/vsk_agentID_param

Take AgentID argument as an object instead of a pointer

Dale-Koenig committed:
CMake install fixes

Dale-Koenig committed:
Remove EOL whitespace

Dale-Koenig committed:
Install headers

Commit:
- removed parameters from behaviours

abhi1287 committed:
Merge pull request #105 from rapyuta-robotics/dk_install_fixes

CMake install fixes

abhi1287 commented on alica_dummy_proxy/src/communication/AlicaDummyCommunication.cpp:
```diff
@@ -124,7 +124,7 @@ void InProcQueue<alica::PlanTreeInfo>::process(std::unique_ptr<alica::PlanTreeIn
     for (AlicaDummyCommunication* module : registeredModules) {
         // Each module expects ownership of shared_ptr
-        auto pti = std::make_shared<alica::PlanTreeInfo>(*first);
+        auto pti = std::make_shared<alica::PlanTreeInfo>((*first).toStandard());
```
Won't the default copy constructor already handle this?

StephanOpfer replied:
I think so! :+1:

abhi1287 commented on alica_engine/include/engine/AlicaContext.h:
```diff
@@ -88,7 +88,7 @@ class AlicaContext
      * @param name Host name
      */
-    static void setRobotName(const std::string& name);
+    static void setLocalAgentName(const std::string& name);
```
Well, the new name does sound better but it can break existing applications. Maybe we can rename it when we try to remove singleton systemconfig and this function won't be static anymore.

StephanOpfer replied:
made method deprecated

abhi1287 commented on alica_engine/include/engine/AlicaEngine.h:
```diff
@@ -128,35 +131,12 @@ class AlicaEngine
-    // TODO: fix this, VariableSyncModule has circular dependency with engine header
-    // VariableSyncModule _variableSyncModule;
-    std::unique_ptr<VariableSyncModule> _variableSyncModule;
+    VariableSyncModule* _variableSyncModule;
```
Reason to get rid of unique_ptr<> ? Ownership aren't clear for raw pointers and also the possibility of leaks.

StephanOpfer replied:
forget to revert it completely, because I tried to make the variableSyncModule a stack variable. Made it a unique_ptr again and added some longer comment about the issue.

abhi1287 commented on alica_engine/include/engine/syncmodule/SyncRow.h:
```diff
@@ -14,23 +14,21 @@ struct SyncData;
-    SyncRow(const SyncData& sd);
+    SyncRow() = delete;
+    explicit SyncRow(const SyncData& sd);
     virtual ~SyncRow();
```
looks like the destructor doesn't need to be virtual

StephanOpfer replied:
I make destructors virtual by default. That way, I don't run into problems when I add a derived class later on. Should we check all classes, whether the have some derived class and remove the virtual keyword, if they don't have one? 
Otherwise, I don't get the point. :sweat_smile:

abhi1287 commented on alica_engine/src/engine/AlicaContext.cpp:
```diff
@@ -17,10 +17,10 @@ constexpr int ALICA_LOOP_TIME_ESTIMATE = 33; // ms
         , _communicator(nullptr)
         , _idManager(std::make_unique<essentials::IDManager>())
         , _engine(std::make_unique<AlicaEngine>(*this, roleSetName, masterPlanName, stepEngine))
-        , _clock(std::make_unique<AlicaClock>())
```
just fyi: initialization order only depends on the declaration order in class.

StephanOpfer replied:
did learn that the hard way ;-)

abhi1287 commented on alica_engine/src/engine/syncmodule/SyncModule.cpp:
```diff
@@ -48,137 +49,118 @@ void SyncModule::init()
+                i->second->integrateSyncTalk(st, _ticks);
             } else {
-                syncProc = new SynchronisationProcess(_ae, _myId, synchronisation, this);
-                _synchSet.insert(pair<const Synchronisation*, SynchronisationProcess*>(synchronisation, syncProc));
```
todo: sync on the protocol

StephanOpfer replied:
:+1:

abhi1287 commented on alica_engine/src/engine/syncmodule/SyncModule.cpp:
```diff
@@ -189,25 +171,57 @@ void SyncModule::onSyncTalk(shared_ptr<SyncTalk> st)
         lock_guard<mutex> lock(_lomutex);
-        map<const Synchronisation*, SynchronisationProcess*>::iterator i = _synchSet.find(synchronisation);
-        if (i != _synchSet.end()) {
+        map<const Synchronisation*, SynchronisationProcess*>::iterator i = _synchProcessMapping.find(synchronisation);
```
as per convention, we are free to use auto for stl iterators.

StephanOpfer replied:
:+1:

abhi1287 commented on alica_engine/src/engine/teammanager/TeamManager.cpp:
```diff
@@ -89,19 +90,18 @@ void TeamManager::readSelfFromConfig()
     constexpr auto notAValidID = std::numeric_limits<uint64_t>::max();
     uint64_t id = sc["Local"]->tryGet<uint64_t>(notAValidID, "Local", "ID", NULL);
     if (id != notAValidID) {
-        std::cout << "[TeamManager] " << id << std::endl;
```
Good to have a final log stating which id agent is going to use, regardless of whether it was from local.conf or autogenerated.

StephanOpfer replied:
added 
    std::cout << "[TeamManager] Own ID is " << _localAnnouncement.senderID << std::endl; as last line in constructor

abhi1287 commented on alica_tests/CMakeLists.txt:
```diff
@@ -1,11 +1,8 @@
-	set(CMAKE_ECLIPSE_VERSION "4.5" CACHE STRING "Eclipse version" FORCE)
-	add_definitions (-DCMAKE_ECLIPSE_GENERATE_SOURCE_PROJECT=TRUE)
-endif (${CMAKE_EXTRA_GENERATOR} MATCHES "Eclipse CDT4")
+add_compile_options(-std=c++14)
```
we are on 17 now.

StephanOpfer replied:
for all packages?

abhi1287 commented on alica_tests/autogenerated/include/Behaviour/NotToTrigger.h:
```diff
@@ -0,0 +1,35 @@
+#pragma once
```
I am assuming these files are just moved to the new location. No file changes other than # pragma

StephanOpfer replied:
:+1:

Commit:
- ported json format without confs to json format with confs

Commit:
- just formatting

Commit:
- fixed doubled included variable bindings

Commit:
- implemented Configuration, ConfAbstractPlanWrapper
- removed artifact from PlanningProblem class (some forward declaration and header files)
- extended parser of the engine for the ConfAbstractWrapper entries in json files

Commit:
- added plan for testing configurations (WIP)
- fixed debugoutput include in header file of query (makes problems elsewhere)

Commit:
- added states into sub plans (fix for invalid assignment problem in task assignment)
- adapted behaviourpool to init basic behaviours with configurations

Commit:
- implemented unit test for configrations on behaviours
- adapted running plan for storing configurations for plantypes and plans, too

Commit:
- finished unit test for reading configurations (now also for plans and plantypes)

Commit:
- removed unused variable

Commit:
- removed conversion via toStandard, because default copy constructor does the job

Commit:
- changed c++ std in cmake lists
- added std::cout in TeamManager constructor for having the final local agent id in the log files
- made getRobotName deprecated in alica context
- changed variable sync module from pointer to unique_ptr in engine

Commit:
- changed c++ std to c++17

Dale-Koenig committed:
Fix flags to prevent warning spam

abhi1287 committed:
Merge pull request #108 from rapyuta-robotics/dk/fix_warning_spam

Fix flags to prevent warning spam

StephanOpfer committed:
Update AlicaContext.h

added message to deprecated method declaration

Commit:
- add missing brace

Commit:
Merge branch 'repair-tests' into featureCommonConfs

Commit:
Merge branch 'v0.9.0' into featureCommonConfs

Commit:
fixed merge issues

StephanOpfer committed:
Merge pull request #107 from rapyuta-robotics/featureCommonConfs

Feature common confs
